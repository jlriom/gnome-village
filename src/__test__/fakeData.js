export const category = {
    category: "category",
    count: 0
}

export const habitantsByHairColor = [
    {
        hair_color: 'Black',
        count: 256
    },
    {
        hair_color: 'Gray',
        count: 258
    },
    {
        hair_color: 'Green',
        count: 267
    },
    {
        hair_color: 'Pink',
        count: 285
    },
    {
        hair_color: 'Red',
        count: 273
    }
];

export const habitantsByProfession = [
    {
        profession: 'Baker',
        count: 214
    },
    {
        profession: 'Blacksmith',
        count: 180
    },
    {
        profession: 'Brewer',
        count: 200
    },
    {
        profession: 'Butcher',
        count: 210
    },
    {
        profession: 'Carpenter',
        count: 216
    },
    {
        profession: 'Cook',
        count: 190
    },
    {
        profession: 'Farmer',
        count: 205
    },
    {
        profession: 'Gemcutter',
        count: 186
    },
    {
        profession: 'Leatherworker',
        count: 185
    },
    {
        profession: 'Marble Carver',
        count: 190
    },
    {
        profession: 'Mason',
        count: 214
    },
    {
        profession: 'Mechanic',
        count: 182
    },
    {
        profession: 'Medic',
        count: 187
    },
    {
        profession: 'Metalworker',
        count: 192
    },
    {
        profession: 'Miner',
        count: 201
    },
    {
        profession: 'Potter',
        count: 202
    },
    {
        profession: 'Prospector',
        count: 202
    },
    {
        profession: 'Sculptor',
        count: 182
    },
    {
        profession: 'Smelter',
        count: 203
    },
    {
        profession: 'Stonecarver',
        count: 187
    },
    {
        profession: 'Tailor',
        count: 199
    },
    {
        profession: 'Tax inspector',
        count: 192
    },
    {
        profession: ' Tinker',
        count: 202
    },
    {
        profession: 'Woodcarver',
        count: 199
    }
];

export const auth = { name: "name", isGuest: false };

export const habitant = {
    id: 362,
    name: 'Cogwitz Bittergrill',
    thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
    age: 127,
    weight: 40.074535,
    height: 93.66314,
    hair_color: 'Black',
    professions: [
        'Prospector',
        ' Tinker',
        'Baker',
        'Tailor',
        'Leatherworker',
        'Sculptor'
    ],
    friends: [
        'Whitwright Fussbuster',
        'Fizwood Bitterrocket',
        'Fizwood Felpower'
    ]
};

export const friends = [
    {
        id: 363,
        name: 'Cogwitzs friend ',
        thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
        age: 127,
        weight: 40.074535,
        height: 93.66314,
        hair_color: 'Black',
        professions: [
            'Prospector',
            ' Tinker'
        ],
        friends: [
            'Whitwright Fussbuster'
        ]
    }
];
