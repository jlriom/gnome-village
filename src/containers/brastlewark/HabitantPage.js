import React, { useEffect } from "react";
import PropTypes from "prop-types";
import { useParams } from "react-router-dom";
import { connect } from "react-redux";
import { loadProfile, loadFriends } from "../../store/actions/userActions";
import HabitantDetails from "../../components/HabitantDetails";

const HabitantPage = ({ id, profile, friends, loadProfile, loadFriends }) => {
	let { name } = useParams();

	useEffect(() => {
		loadProfile(name);
	}, [loadProfile, name]);

	useEffect(() => {
		loadFriends(name);
	}, [loadFriends, name]);

	return <HabitantDetails profile={profile} friends={friends} />;
};

HabitantPage.propTypes = {
	id: PropTypes.number,
	profile: PropTypes.object.isRequired,
	friends: PropTypes.array.isRequired,
	loadProfile: PropTypes.func.isRequired,
	loadFriends: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
	id: state.user.id,
	profile: state.user.profile,
	friends: state.user.friends
});

export default connect(
	mapStateToProps,
	{ loadProfile, loadFriends }
)(HabitantPage);
