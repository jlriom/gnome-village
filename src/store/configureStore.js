import { createStore, applyMiddleware, compose } from "redux";
import reduxImmutableStateInvariant from "redux-immutable-state-invariant";
import thunk from "redux-thunk";
import logger from "redux-logger";

import rootReducer from "./reducers";

const middlewares = [thunk, logger, reduxImmutableStateInvariant()];

export default function configureStore(initialState) {
	const composeEnhancers =
		window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; // add support for Redux dev tools

	return createStore(
		rootReducer,
		initialState,
		composeEnhancers(applyMiddleware(...middlewares))
	);
}
