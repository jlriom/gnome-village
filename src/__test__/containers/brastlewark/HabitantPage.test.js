import React from "react";
import HabitantPage from "../../../containers/brastlewark/HabitantPage";
import renderer from "react-test-renderer";
import { MemoryRouter } from 'react-router-dom';
import { habitant, friends } from "../../fakeData";
import { Provider } from "react-redux";
import configureStore from "../../../store/configureStore";
const store = configureStore();

const props = {
    id: 0,
    profile: habitant,
    friends: friends,
    loadProfile: jest.fn(),
    loadFriends: jest.fn()
}

describe("<HabitantPage />", () => {
    it("renders ok", () => {
        const tree = renderer.create(
            <Provider store={store}>
                <MemoryRouter>
                    <HabitantPage {...props} />
                </MemoryRouter>
            </Provider>
        );
        expect(tree).toMatchSnapshot();
    });
});
