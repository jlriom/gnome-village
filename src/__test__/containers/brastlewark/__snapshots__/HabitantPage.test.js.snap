// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HabitantPage /> renders ok 1`] = `
Array [
  "",
  "",
  <div>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-3"
        >
          <div
            className="card"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <div
              className="lazyload-placeholder"
              style={
                Object {
                  "height": undefined,
                }
              }
            />
            <div
              className="card-body"
            >
              <div
                className="card-title h5"
              >
                
              </div>
              <div
                className="container"
                style={
                  Object {
                    "weight": "100%",
                  }
                }
              >
                <dl
                  className="row"
                >
                  <dt
                    className="col-sm-6"
                  >
                     
                    Age:
                  </dt>
                  <dd
                    className="col-sm-6"
                  >
                     
                    0
                  </dd>
                  <dt
                    className="col-sm-6"
                  >
                     
                    Hair Color:
                  </dt>
                  <dd
                    className="col-sm-6"
                  >
                     
                    Black
                  </dd>
                  <dt
                    className="col-sm-6"
                  >
                     
                    Weight:
                  </dt>
                  <dd
                    className="col-sm-6"
                  >
                     
                    0.000
                  </dd>
                  <dt
                    className="col-sm-6"
                  >
                     
                    Height:
                  </dt>
                  <dd
                    className="col-sm-6"
                  >
                     
                    0.000
                  </dd>
                </dl>
                <hr />
                <div
                  className="row"
                >
                  <span />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-9"
        >
          <h3
            className="m-4"
          >
            Friends
          </h3>
          <br />
        </div>
      </div>
    </div>
  </div>,
]
`;
